aws-api/src/test/java/com/amplifyframework/api/aws/auth/OwnerBasedAuthTest.java

```diff
@@ -0,0 +1,351 @@
+    private static final String GRAPHQL_API_WITH_OIDC = "graphQlApi_oidc";
+
+    private MockWebServer webServer;
+    private HttpUrl baseUrl;
+    private AWSApiPlugin plugin;
+    private CognitoUserPoolsAuthProvider cognitoProvider;
+    private OidcAuthProvider oidcProvider;
+    private String apiName;
+
+    /**
+     * Sets up the test.
+     * @throws ApiException On failure to configure plugin
+     * @throws IOException On failure to start web server
+     */
+    @Before
+    public void setup() throws ApiException, IOException {
+        webServer = new MockWebServer();
+        webServer.start(8080);
+        baseUrl = webServer.url("/");
+        cognitoProvider = new FakeCognitoAuthProvider();
+        oidcProvider = new FakeOidcAuthProvider();
+        configurePlugin();
+    }
+
+    /**
+     * Stop the {@link MockWebServer} that was started in {@link #setup()}.
+     * @throws IOException On failure to shutdown the MockWebServer
+     */
+    @After
+    public void cleanup() throws IOException {
+        webServer.shutdown();
+    }
+
+    private void configurePlugin() throws ApiException {
+        ApiAuthProviders providers = ApiAuthProviders.builder()
+                .cognitoUserPoolsAuthProvider(cognitoProvider)
+                .oidcAuthProvider(oidcProvider)
+                .build();
+        JSONObject configuration = new JSONObject();
+        try {
+            configuration = configuration
+                .put(GRAPHQL_API_WITH_API_KEY, new JSONObject()
+                        .put("endpointType", "GraphQL")
+                        .put("endpoint", baseUrl.url())
+                        .put("region", "us-east-1")
+                        .put("authorizationType", "API_KEY")
+                        .put("apiKey", "FAKE-API-KEY"))
+                .put(GRAPHQL_API_WITH_COGNITO, new JSONObject()
+                        .put("endpointType", "GraphQL")
+                        .put("endpoint", baseUrl.url())
+                        .put("region", "us-east-1")
+                        .put("authorizationType", "AMAZON_COGNITO_USER_POOLS"))
+                .put(GRAPHQL_API_WITH_OIDC, new JSONObject()
+                        .put("endpointType", "GraphQL")
+                        .put("endpoint", baseUrl.url())
+                        .put("region", "us-east-1")
+                        .put("authorizationType", "OPENID_CONNECT"));
+        } catch (JSONException exception) {
+            // This shouldn't happen...
+        }
+
+        plugin = new AWSApiPlugin(providers);
+        plugin.configure(configuration, ApplicationProvider.getApplicationContext());
+    }
+
+    /**
+     * Test that owner argument fails to be appended to subscription request if
+     * the authorization mode is not OIDC compliant.
+     */
+    @Test
+    public void ownerArgumentNotAddedWithApiKey() {
+        // Set API to use API key auth mode
+        apiName = GRAPHQL_API_WITH_API_KEY;
```

Comment:
This is a little weird, where you set the `apiName`, but then `apiName` isn't used anywhere in this `@Test`. IF you need the `apiName` value elsewhere, can you pass it into a helper method, instance of relying on the instance state of the test class?

Reply:
I do agree that relying on the instance state is less than ideal, and I did give this approach a try. I decided against it because
1. most of these tests were not originally written by me, so I would rather maintain the syntax of the original tests when appending my own.
2. adding in another parameter to `isOwnerArgumentAdded()` method made the test suite look overall very verbose without adding too much value to the comprehensibility of the tests (since every assertion in a given test case will use the same API).